products/compute/api.yaml

```diff
@@ -2205,12 +2204,15 @@ objects:
         description: |
           The purpose of the resource. For global internal addresses it can be
 
-          * VPC_PEERING - for peer networks
-
-          This should only be set when using an Internal address.
+          * GCE_ENDPOINT for addresses that are used by VM instances, alias IP ranges, internal load balancers, and similar resources.
+          * DNS_RESOLVER for a DNS resolver address in a subnetwork
+          * VPC_PEERING for addresses that are reserved for VPC peer networks.
+          * NAT_AUTO for addresses that are external IP addresses automatically reserved for Cloud NAT.
         values:
+        - :GCE_ENDPOINT
+        - :DNS_RESOLVER
```

Comment:
When I originally added this these options weren't fully supported yet. I'm assuming if they are documented in the GA docs then they should be working but might want to double check.